docs/changes/config/17934.bugfix.rst

```diff
@@ -0,0 +1,5 @@
+``get_config_dir()`` and ``get_cache_dir()`` now warn about non-existent
+directories specified via ``XDG_CACHE_HOME`` and ``XDG_CONFIG_HOME`` environment
+variables respectively. Missing sub-directories (i.e.
+``XDG_(CACHE|CONFIG)_HOME/astropy``) will also be silently created as long as the
```

Comment:
But this part I still don't quite get it. On `main`, it silently creates it anyway, only it is created as symlink via linkto, so that it is consistent with its own logic of simply deferring back to `cls._find_or_create_root_dir(linkto=None, pkgname=rootname)` in the event of the symlink already exist.

Maybe we don't need the second change here?

Reply:
In testing the `main` code, I found the following behavior:

1. The symlink only happens if `maindir` (e.g., `/home/user/.astropy/config`) does not exist and `$XDG_CONFIG_HOME` is set but `$XDG_CONFIG_HOME/astropy` does not exist.
2. If the condition above is met, the symlink happens for "config" as soon as astropy is imported. This surprised me but then it makes sense because astropy needs to know where to grab the config for itself on import.
3. However, the "cache" symlink does not happen until it is needed (usually the download stuff).

I find all the monkeypatching hard to follow, so I prefer to test its "real world" behavior and sort it out before I focus on the tests.

Reply:
> I find all the monkeypatching hard to follow, so I prefer to test its "real world" behavior and sort it out before I focus on the tests.

fair enough.
Everything you found seems okay to me, do you see a problem there ?